File: app_launcher/src/main/java/com/pandaq/app_launcher/ui/framework/AppBaseActivity.java
package com.pandaq.app_launcher.ui.framework;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.Toolbar;
import android.widget.FrameLayout;

import com.pandaq.app_launcher.R;
import com.pandaq.appcore.framework.mvp.BaseActivity;
import com.pandaq.appcore.framework.mvp.BasePresenter;
import com.pandaq.uires.guide.GuideCoverView;
import com.pandaq.uires.msgwindow.Toaster;
import com.pandaq.uires.progress.LoadingDialog;

/**
 * Created by huxinyu on 2018/1/26.
 * Email : panda.h@foxmail.com
 * Description :所有 BaseActivity 类的最基础类
 */

public abstract class AppBaseActivity<P extends BasePresenter> extends BaseActivity<P> {

    protected @Nullable
    Toolbar mToolbar;
    private FrameLayout mParentView;
    /**
     * 遮罩引导载体图层
     */
    private GuideCoverView mGuideCoverView;
    /**
     * 标识 activity 是否是向导 BaseActivity
     */
    protected boolean isGuide;

    private LoadingDialog mLoadingDialog;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // 如果是新手向导页则初始化向导载体图层
        if (isGuide) {
            initGuide();
        }
        initToolBar();
    }

    /**
     * 初始化ToolBar
     */
    public void initToolBar() {
        mToolbar = findViewById(R.id.titleBar);
        if (mToolbar != null) {
            setSupportActionBar(mToolbar);
            mToolbar.setNavigationOnClickListener(v -> onBackPressed());
        }
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        setIntent(intent);
    }

    /**
     * 初始化向导图层
     */
    private void initGuide() {
        mParentView = (FrameLayout) getWindow().getDecorView();
        if (mGuideCoverView == null) {
            mGuideCoverView = new GuideCoverView(mParentView);
        }
    }

    /**
     * 显示 GuideCoverV
     */
    protected void showGuideView() {
        if (mGuideCoverView != null) {
            mParentView.addView(mGuideCoverView);
        }
    }

    public void showLoading(String msg) {
        if (mLoadingDialog == null) {
            mLoadingDialog = new LoadingDialog(this);
        }
        mLoadingDialog.show(false);
    }

    public void hideLoading() {
        if (mLoadingDialog != null) {
            mLoadingDialog.dismiss();
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        hideLoading();
        mLoadingDialog = null;
    }

    public void onError(long errCode, String errMsg) {
        Toaster.msg(errMsg)
                .show();
    }

    public void onLoadFinish() {
        // 加载完成肯定会调用隐藏 loading
        hideLoading();
    }


}

File: app_launcher/src/main/java/com/pandaq/app_launcher/ui/framework/AppBaseActivity.kt
package com.pandaq.app_launcher.ui.framework

import android.os.Bundle
import android.support.v7.app.ActionBar
import android.view.View
import android.view.WindowManager
import com.pandaq.app_launcher.R
import com.pandaq.appcore.framework.app.ActivityTask
import com.pandaq.appcore.framework.mvp.BaseActivity
import com.pandaq.uires.loading.LoadingDialogUtil
import com.pandaq.uires.msgwindow.Toaster
import com.pandaq.uires.widget.toolbar.CNToolbar


/**
 * Created by huxinyu on 2019/7/8.
 * Email : panda.h@foxmail.com
 * <p>
 * Description :
 */
abstract class AppBaseActivity<P : AppBasePresenter<*>> : BaseActivity<P>() {

    protected var mToolbar: CNToolbar? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
    }

    fun setTitle(title: String) {
        mToolbar?.setTitle(title)
    }

    override fun initToolbar() {
        val view = layoutInflater.inflate(R.layout.res_cn_title, null)
        mToolbar = view.findViewById(R.id.toolbar)
        mToolbar?.let {
            it.setOnBackPressed(View.OnClickListener { onBackPressed() })
            it.setTitle(title)
        }
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.title = ""
            actionBar.setDisplayShowHomeEnabled(false)
            actionBar.setDisplayShowCustomEnabled(true)
            val alp =
                    ActionBar.LayoutParams(ActionBar.LayoutParams.MATCH_PARENT, ActionBar.LayoutParams.MATCH_PARENT)
            actionBar.setCustomView(view, alp)
        }
    }

    override fun showLoading(msg: String?) {
        LoadingDialogUtil.show(ActivityTask.getInstance().topActivity, msg, true)
    }

    override fun showLoading() {
        LoadingDialogUtil.show(ActivityTask.getInstance().topActivity, true)
    }

    override fun hideLoading() {
        LoadingDialogUtil.hideProgressQuick()
    }

    override fun showLoading(cancelAble: Boolean) {
        LoadingDialogUtil.show(ActivityTask.getInstance().topActivity, cancelAble)
    }

    override fun onError(errCode: Long, errMsg: String?) {
        errMsg?.let {
            Toaster.msg(errMsg).showError()
        }
    }

    override fun onLoadFinish(success: Boolean) {
        hideLoading()
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        outState?.putParcelable("android:support:fragments", null)
    }
}